src/Synth/OscilGen.h

```diff
@@ -14,18 +14,19 @@
 #ifndef OSCIL_GEN_H
 #define OSCIL_GEN_H
 
+#include "../Misc/ClassWithPorts.h"
 #include "../globals.h"
 #include <rtosc/ports.h>
 #include "../Params/Presets.h"
 #include "../Params/WaveTableFwd.h"
 
 namespace zyn {
 
-class OscilGen:public Presets
+class OscilGen:public Presets, public ClassWithPorts
```

Comment:
Perhaps a dumb question, but making 'Presets' an instance of ClassWithPorts would avoid the multiple inheritance which caused you some headaches. Would that simply things at all? (feel free to ignore if it complicates anything)

Reply:
Should work, but it would force me to add virtuals to some other classes (e.g. `SUBnoteParameters`) that currently don't seem to use that non-RT PresetExtractor stuff.

Anyways, there's a chance we replace `ClassWithPorts` with a different approach, like written in my comment about/in 905f238.